### ISSUE TITLE
Godot 4.3 beta 1 Parser Errors that don't appear in Godot 4.2.2

### ISSUE THREAD

**Issue by svendixon** [godotengine/godot#92780]
### Tested versions

Godot 4.3 beta 1

### System information

Windows 11, Vulkan, GTX 980 Ti (545.84)

### Issue description

An autoload "PreloadedScenes.gd" that preloads a scene "some_node.tscn".
```gdscript
extends Node
const SOME_NODE := preload("res://some_node.tscn")
```
"some_node.tscn" has an attached script with a custom class name and var of type "Game" which is another custom class name for the game scene.
```gdscript
class_name SomeNode
extends Node
var game : Game
```
The game scene instantiates "some_node". Gives the error: `Parser Error: Could not resolve member "SOME_NODE".`
```gdscript
class_name Game
extends Node2D
var some_node := PreloadedScenes.SOME_NODE.instantiate() as SomeNode
```
Even when preloading it directly, it gives the error: `Parser Error: Could not preload resource file "res://some_node.tscn".`
```gdscript
class_name Game
extends Node2D
var some_node := preload("res://some_node.tscn").instantiate() as SomeNode
```
I'm not an experienced programmer so I'm unsure what is happening here. I assume this is a cyclic reference problem that Godot 4.2.2 was able to deal with but Godot 4.3 beta 1 no longer can?

### Steps to reproduce

Open the minimal reproduction project and run it. 

If you open it in 4.2.2, it will run. If you open it in 4.3 beta 1, it will error.

### Minimal reproduction project (MRP)

[SomeNode.zip](https://github.com/user-attachments/files/15573196/SomeNode.zip)

**Comment by Saul2022** [comment]
I think i have the same issue but was  from being in 4.3 dev 6 to 4.3 beta 1, could you try checking in 4.3 dev 6 if it works ? Though i didn’t use global, just loading screen thing with the main scene.

**Comment by matheusmdx** [comment]
Reverting https://github.com/godotengine/godot/pull/85501 on latest master [71699e0] fixes the problem, also this looks similar to #90954 and pr #92326 fixes #90954 but doesn't fix this pr

**Comment by anniryynanen** [comment]
This looks fun, I'll see if I can do something about it.

**Comment by anniryynanen** [comment]
#93346 fixes this, the error doesn't happen in v4.3.beta.gh-93346 [c5b5b90df].
